### DISCUSSION TITLE
Drawing directed edges on a mesh?

### DISCUSSION THREAD

**Discussion by rahul-mitra13** [libigl/libigl#2050] [Q&A]
Hello, 

I'm attempting to use libigl to draw a directed graph on a mesh. So far, I've been using the add_edges() method, which draws undirected edges. Was wondering if there's any code I could add/use/modify to draw directed edges instead? Would appreciate any help. 

Rahul

**Comment by alecjacobson** [comment]
do you mean that you want to draw little arrows?

**Reply by rahul-mitra13** [reply]
Yep.

**Reply by rahul-mitra13** [reply]
Was wondering if there's any follow-up on this? Any thread I could reference? Thanks!

**Reply by jdumas** [reply]
There's an old pull request #875 with some shader code to draw arrows, but it's never been integrated to the viewer. So for now there is no official support for directed edge drawing.

**Reply by alecjacobson** [reply]
right. I think it's unlikely that we integrated fancy arrows into the viewer. Hacking together simple ones via edges could be a tiny bit of code, if it's good enough. Meanwhile, if you need nice arrows, I'd recommend polyscope

**Reply by rahul-mitra13** [reply]
Gotcha, thanks for the follow-up. Will look into the suggestions.
